source/write/transactions.txt

```diff
@@ -0,0 +1,160 @@
+that change data only if the entire transaction is committed.
+If any operation in the transaction does not succeed, the driver stops the
+transaction and discards all data changes before they ever become
+visible. This feature is called **atomicity**.
+
+In MongoDB, transactions run within logical **sessions**. A
+session is a grouping of related read or write operations that you
+want to run sequentially. Sessions enable causal consistency for a group
+of operations and allow you to run operations in an **ACID-compliant**
+transaction, which is a transaction that meets an expectation of
+atomicity, consistency, isolation, and durability. MongoDB guarantees
+that the data involved in your transaction operations remains
+consistent, even if the operations encounter unexpected errors.
+
+When using the {+driver-short+}, you can start a ``ClientSession`` by
+calling the ``startSession()`` method on your client. Then, you can
+perform transactions within the session.
+
+.. warning::
+
+   Use a ``ClientSession`` only in operations running on the
+   ``MongoClient`` that created it. Using a ``ClientSession`` with a
+   different ``MongoClient`` results in operation errors.
+
+Methods
+-------
+
+After calling the ``startSession()`` method to start a session, you can
+use methods from the ``ClientSession`` class to modify the session state.
+The following table describes the methods you can use to manage a
+transaction:
+
+.. list-table::
+   :widths: 25 75
+   :stub-columns: 1
+   :header-rows: 1
+
+   * - Method
+     - Description
+
+   * - ``startTransaction()``
+     - | Starts a new transaction on this session. You cannot start a
+         transaction if there's already an active transaction running in
+         the session.
+       |
+       | You can set transaction options by passing a ``TransactionOptions``
+         instance as a parameter.
+
+   * - ``commitTransaction()``
+     - | Commits the active transaction for this session. This method returns an
+         error if there is no active transaction for the session, the
+         transaction was previously ended, or if there is a write conflict.
+
+   * - ``abortTransaction()``
+     - | Ends the active transaction for this session. This method returns an
+         error if there is no active transaction for the session or if the
+         transaction was committed or ended.
+
+.. _scala-txn-example:
+
+Transaction Example
+-------------------
+
+This example defines a ``runTransaction()`` method that
+modifies data in the collections of the ``sample_mflix`` database.
+The code performs the following actions:
+
+- Creates ``MongoCollection`` instances to access the ``movies``
+  and ``users`` collections.
+- Specifies the read preference, read concern, and write concern for
+  the transaction.
+- Starts the transaction.
+- Inserts a document into the ``movies`` collection and prints the results.
+- Updates a document in the ``users`` collection and prints the results.
```

Comment:
Since these aren't full sentences
```suggestion
- Creates ``MongoCollection`` instances to access the ``movies``
  and ``users`` collections
- Specifies the read preference, read concern, and write concern for
  the transaction
- Starts the transaction
- Inserts a document into the ``movies`` collection and prints the results
- Updates a document in the ``users`` collection and prints the results
```